lark/grammars/python.lark

```diff
@@ -228,14 +228,15 @@ COMMENT: /#[^\n]*/
 STRING : /[ubf]?r?("(?!"").*?(?<!\\)(\\\\)*?"|'(?!'').*?(?<!\\)(\\\\)*?')/i
 LONG_STRING: /[ubf]?r?(""".*?(?<!\\)(\\\\)*?"""|'''.*?(?<!\\)(\\\\)*?''')/is
 
-DEC_NUMBER: /0|[1-9][\d_]*/i
+DEC_NUMBER: /0|\d(?:_?\d)*/i
 HEX_NUMBER.2: /0x[\da-f]*/i
 OCT_NUMBER.2: /0o[0-7]*/i
-BIN_NUMBER.2 : /0b[0-1]*/i
-FLOAT_NUMBER.2: /((\d+\.[\d_]*|\.[\d_]+)([Ee][-+]?\d+)?|\d+([Ee][-+]?\d+))/
-IMAG_NUMBER.2: /\d+[Jj]/ | FLOAT_NUMBER /[Jj]/
+BIN_NUMBER.2: /0b[0-1]*/i
```

Comment:
This pattern is also wrong, same for `OCT` and `HEX` as well.

Reply:
Hmmm I remember I was testing if I could insert `_` into the bin and I failed (I extrapolated this knowledge to oct and hex), but now I checked again and everything is fine
```python
>>> 0b1_0
2
```
I don't even know how I couldn't write that before :)

I will fix it, thanks!

Reply:
And that's interesting that we can do 
```python
>>> 0b_10
2
>>> 0x_10
16
```

Reply:
resolved in 8a67efbd5ede2ec2bc76e626bb93dfecd5f16ab9